patching/config/test/getPatchInfo.test.ts

```diff
@@ -0,0 +1,163 @@
+import { getPatchInfo } from '../src/getPatchInfo'
+import { type PatchGroupRecord } from '../src/index'
+
+test('getPatchInfo(undefined, ...) returns undefined', () => {
+  expect(getPatchInfo(undefined, 'foo', '1.0.0')).toBeUndefined()
+})
+
+test('getPatchInfo() returns exact version if match', () => {
+  const patchedDependencies = {
+    foo: {
+      exact: {
+        '1.0.0': {
+          file: {
+            path: 'patches/foo@1.0.0.patch',
+            hash: '00000000000000000000000000000000',
+          },
+          key: 'foo@1.0.0',
+          strict: true,
+        },
+      },
+      range: {},
+      blank: undefined,
+    },
+  } satisfies PatchGroupRecord
+  expect(getPatchInfo(patchedDependencies, 'foo', '1.0.0')).toStrictEqual(patchedDependencies.foo.exact['1.0.0'])
+  expect(getPatchInfo(patchedDependencies, 'foo', '1.1.0')).toBeUndefined()
+  expect(getPatchInfo(patchedDependencies, 'foo', '2.0.0')).toBeUndefined()
+  expect(getPatchInfo(patchedDependencies, 'bar', '1.0.0')).toBeUndefined()
+})
+
+test('getPatchInfo() returns range version if satisfied', () => {
+  const patchedDependencies = {
+    foo: {
+      exact: {},
+      range: {
+        1: {
+          file: {
+            path: 'patches/foo@1.patch',
+            hash: '00000000000000000000000000000000',
+          },
+          key: 'foo@1',
+          strict: true,
+        },
+      },
+      blank: undefined,
+    },
+  } satisfies PatchGroupRecord
+  expect(getPatchInfo(patchedDependencies, 'foo', '1.0.0')).toStrictEqual(patchedDependencies.foo.range[1])
```

Comment:
why is it a number? it should be a string
```suggestion
  expect(getPatchInfo(patchedDependencies, 'foo', '1.0.0')).toStrictEqual(patchedDependencies.foo.range['1'])
```

Reply:
https://github.com/pnpm/pnpm/pull/9236/commits/55e44ddd1dbb8bd467de2f77ab29a81a0f9f25fc